core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -51,8 +54,39 @@ private[spark] class TaskDescription(
     val index: Int,    // Index within this task's TaskSet
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
-    val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    val properties: Properties) extends Logging {
+
+  def this(
+      taskId: Long,
+      attemptNumber: Int,
+      executorId: String,
+      name: String,
+      index: Int, // Index within this task's TaskSet
+      addedFiles: Map[String, Long],
+      addedJars: Map[String, Long],
+      properties: Properties,
+      task: Task[_]) {
+      this(taskId, attemptNumber, executorId, name, index,
+        addedFiles, addedJars, properties)
+      task_ = task
```

Comment:
Honestly, don't totally remember my earlier comment -- I think I may also have been grasping at an attempt to make this easier to follow.  Your suggestion sounds at least as good as other versions, so its fine with me.